###  ques--1 print the even number with continue statement

In [1]:
n=int(input("enter the number where we end even number : "))
for i in range(2,n+1,2):
    if i==10:
        continue
    print(i)

enter the number where we end even number : 5
2
4


##  ques --2 Write a python program to accept number from user and check the if it is prime or not"

In [2]:
n=int(input())
if n > 1:
    for i in range(2,n+1):
        if (n%i) == 0:
            print(n," not is a prime number")
            break
    else:
        print(n," is a prime number")

12
12  not is a prime number


### " ques--3 Write a python program to print the first 10 numbers Fibonacci series"

In [3]:
n1=0
n2=1
print("Fibonacci series")
print(n1, n2,end=",")
for i in range(2, 10):
    fseg= n1 + n2
    print(fseg,end = ",")
    n1,n2= n2,fseg


Fibonacci series
0 1,1,2,3,5,8,13,21,34,

### " ques--4 Write a program to print the following series till terms 4,44,444china …..n terms

In [4]:
def print_series(n):
    term = 4
    i = 0
    while i < n:
        # Check if the current iteration count is less than n
        if i < n:
            print(term)
            # Use a for loop to construct the next term
            for _ in range(i):
                term = term * 10 + 4
        i += 1

# Example usage: print the series up to 5 terms
print_series(5)

4
4
44
4444
4444444


###  ques--5 Write a program to print multiplication table of 7 using break statement."


In [5]:
n=7
print("Table of ",n)
for i in range (1,11):
    num=n*i
    
    print("7*",i,"=",num)
    if num==n*10:
        break
        
        


Table of  7
7* 1 = 7
7* 2 = 14
7* 3 = 21
7* 4 = 28
7* 5 = 35
7* 6 = 42
7* 7 = 49
7* 8 = 56
7* 9 = 63
7* 10 = 70


### ques-- 6 Write a program to determine how many of the numbers between 1 and 10000 contain the digit 3."

In [18]:
count=0
number=1
while number <=10000:
    for digit in str(number):
        if digit == '3':
            count +=1
            break
    number +=1
print(count)


3439


### ques-- 7 Given a string s, s.index('x') returns the index of the first x in s and an error if there is no x.((a) Write a program that asks the user for a string and a letter. Using a while loop, the program should print the index of the first occurrence of that letter and a message if the string does not contain the letter.) ((b) Write the above program using a for/break loop instead of a while loop)"

In [9]:
string=input("enter a string : ")
letter=input("enter a letter : ")
index=0
found=False
while index<len(string):
    if string[index]== letter:
        found=True
        break
    index+=1
if found:
    print(f"the index of the frist string'{letter}'in the string is: {index}")
else:
    print(f"the string does not contain the letter'{letter}'.")

enter a stringanuj
enter a lettera
the index of the frist string'a'in the string is: 0


In [14]:
str1=(input("enter a string ="))
letter=input(" enter a letter")
index=0
found=False
if str1[0].isalpha():
    while index<len(str1):
        if str1[index]==letter:
            found=True
            print({letter},{index})
            break
        index +=1
else:
    print("enter a sentance those strat with alphabates : ")
           
if not found:
    print(letter)

enter a string =123anuj
 enter a lettera
enter a sentance those strat with alphabates : 
a


In [22]:
s = input("Enter a string: ")
letter =input("enter a letter")
a=(len(s))
# For loop to find the letter
if s[0].isalpha():
    for index,char in enumerate(s):
        if char == letter:
            print(f"The index of the first occurrence of the letter '{letter}' is: {index}")
            break
    else:
        print(f"The string does not contain the letter '{letter}'.")
else:
    print("enter a string those start with alphabates ")

Enter a string: 122 anuj
enter a lettera
enter a string those start with alphabates 


### ques--8  Write a program to calculate the sum of first 10 number using for loop."

In [9]:
count = 0

sum = 0

for count in range(1,11):
    print ('count =',count)
    count += 1
    sum += count-1
    
print ('sum of first 10 number : ',sum)

print ('Progrsm completed.')
  

count = 1
count = 2
count = 3
count = 4
count = 5
count = 6
count = 7
count = 8
count = 9
count = 10
sum of first 10 number :  55
Progrsm completed.


### " ques--9 Write a program that computes the factorial of a number. The factorial, n!, of a number n is the product of all the integers between 1 and n, including n."

In [22]:
def factorial(n):
    if n < 0:
        return "Factorial is not defined for negative numbers."
    elif n == 0:
        return 1
    else:
        result = 1
        for i in range(1, n + 1):
            result *= i
        return result

# Example usage:
number = int(input('enter tyhe number'))  # Replace 5 with any non-negative integer
print(f"The factorial of {number} is: {factorial(number)}")

enter tyhe number5
The factorial of 5 is: 120


In [24]:
num = int(input("Enter a number: "))


factorial = 1

for i in range(1, num + 1):
    factorial *= i

print(f"The factorial of {num} is: {factorial}")

print('Program Completed.')

Enter a number: 5
The factorial of 5 is: 120
Program Completed.


### ques-- 10 Adding certain numbers to their reversals sometimes produces a palindromic number. For instance, 241 + 142 = 383. Sometimes, we have to repeat the process. For instance, 84 + 48 = 132 and 132 + 231 = 363. Write a program that finds both two-digit numbers for which this process must be repeated more than 20 times to obtain a palindromic number."

In [8]:
num=input("enter the 2 digits number:")
count=num
while True:
    str_num=str(num)
    rev_num=str_num[::-1]
    if str_num==rev_num:
        print(num,"is a palindrome")
        break
    num=int(str_num)+int(rev_num)
    print(num)

enter the 2 digits number:89
187
968
1837
9218
17347
91718
173437
907808
1716517
8872688
17735476
85189247
159487405
664272356
1317544822
3602001953
7193004016
13297007933
47267087164
93445163438
176881317877
955594506548
1801200002107
8813200023188
8813200023188 is a palindrome


In [7]:

count = 0

for i in range(10, 100):
    num = i
    repeat_count = 0
    while repeat_count <= 20:
        num = num + int(str(num)[::-1])
        if str(num) == str(num)[::-1]:
            break
        repeat_count += 1

    if repeat_count > 20:
        print("Two-digit number:", i)

print("Program completed.")

Two-digit number: 89
Two-digit number: 98
Program completed.


### ques--11  What is the difference between break and continue statement. Provide the solution with coding examples"

In [28]:
total=0
for i in range (1,11):
    total += i
    print(total)
    # but continue statement work with both condition true or flase.it is print the vanlu
    continue

1
3
6
10
15
21
28
36
45
55


In [29]:
total=0
for i in range (1,11):
    total += i
    print(total)
    # if we are use break statement.break statement with a true condition .it will not print the value after true condition just like that
    break
    

1


###  ques-- 12 Write a program to calculate the interest rate is 3%, compound interest of each month increases by 0.4 up to 12 months and loan of 1,00,00."

In [30]:
p=10000
rate=3
time_limit=12
month=1
while month<=time_limit:
    rate+=0.4
    amount=p*(1+rate/100)**month
    ci=amount-p
    print("month : ",month,"--","compound interest : ",round(ci,3))
    month+=1
print("program is completed")

month :  1 -- compound interest :  340.0
month :  2 -- compound interest :  774.44
month :  3 -- compound interest :  1313.661
month :  4 -- compound interest :  1970.898
month :  5 -- compound interest :  2762.816
month :  6 -- compound interest :  3710.196
month :  7 -- compound interest :  4838.83
month :  8 -- compound interest :  6180.656
month :  9 -- compound interest :  7775.214
month :  10 -- compound interest :  9671.514
month :  11 -- compound interest :  11930.427
month :  12 -- compound interest :  14627.772
program is completed


###  ques--13 Write a program that lets the user play Rock-Paper-Scissors against the computer. In that program there were exactly five rounds. Rewrite the program so that it is a best 3 out of 5. That is, the first player to win three times is the winner"

In [31]:
user_score = 0
computer_score = 0
rounds_to_win = 3

print("Let's play Rock-Paper-Scissors! First player to win three times is the winner.")

while user_score < rounds_to_win and computer_score < rounds_to_win:
    round_number = 0
    while round_number < 5:
        user_choice = input("Enter your choice (rock, paper, or scissors): ").lower()
        while user_choice not in ["rock", "paper", "scissors"]:
            print("Invalid choice. Please enter rock, paper, or scissors.")
            user_choice = input("Enter your choice (rock, paper, or scissors): ").lower()

        computer_choice = ["rock", "paper", "scissors"][(round(hash(str(round_number)) * 1000)) % 3]

        print("You chose:", user_choice)
        print("Computer chose:", computer_choice)

        if user_choice == computer_choice:
            print("It's a tie!")
        elif (user_choice == "rock" and computer_choice == "scissors") or \
             (user_choice == "paper" and computer_choice == "rock") or \
             (user_choice == "scissors" and computer_choice == "paper"):
            print("You win!")
            user_score += 1
        else:
            print("Computer wins!")
            computer_score += 1
        
        round_number += 1

if user_score > computer_score:
    print("Congratulations! You won the game!")
else:
    print("Sorry, computer won the game!")
print('Program completed.')

Let's play Rock-Paper-Scissors! First player to win three times is the winner.
Enter your choice (rock, paper, or scissors): paper
You chose: paper
Computer chose: scissors
Computer wins!
Enter your choice (rock, paper, or scissors): rock
You chose: rock
Computer chose: scissors
You win!
Enter your choice (rock, paper, or scissors): scissiors
Invalid choice. Please enter rock, paper, or scissors.
Enter your choice (rock, paper, or scissors): paper
You chose: paper
Computer chose: paper
It's a tie!
Enter your choice (rock, paper, or scissors): rock
You chose: rock
Computer chose: rock
It's a tie!
Enter your choice (rock, paper, or scissors): scissors
You chose: scissors
Computer chose: paper
You win!
Enter your choice (rock, paper, or scissors): rock
You chose: rock
Computer chose: scissors
You win!
Enter your choice (rock, paper, or scissors): rock
You chose: rock
Computer chose: scissors
You win!
Enter your choice (rock, paper, or scissors): rock
You chose: rock
Computer chose: paper


KeyboardInterrupt: Interrupted by user

### ques-- 14  Write a program to print to find the sum of series .1 + 2 + 6 + 24 +120 … n terms."

In [1]:
num=int(input("enter thart number you want find sum of series : "))
sum_series=0
factorial=1
for i in range(1,num+1):
    factorial*=i
    sum_series+=factorial
print(sum_series)

enter thart number you want find sum of series : 5
153


### ques 15 What is the nested loop explain by using an example"

In [3]:
# a loop create with in loop statements is called nested loop
for i in range(1,6):
    for j in range(1,7):
        div=round(i/j,1)
        if div==0.2:
            break
        else:
            print(i,"/",j,"==",div)
print(" program is comppleted")
        

1 / 1 == 1.0
1 / 2 == 0.5
1 / 3 == 0.3
2 / 1 == 2.0
2 / 2 == 1.0
2 / 3 == 0.7
2 / 4 == 0.5
2 / 5 == 0.4
2 / 6 == 0.3
3 / 1 == 3.0
3 / 2 == 1.5
3 / 3 == 1.0
3 / 4 == 0.8
3 / 5 == 0.6
3 / 6 == 0.5
4 / 1 == 4.0
4 / 2 == 2.0
4 / 3 == 1.3
4 / 4 == 1.0
4 / 5 == 0.8
4 / 6 == 0.7
5 / 1 == 5.0
5 / 2 == 2.5
5 / 3 == 1.7
5 / 4 == 1.2
5 / 5 == 1.0
5 / 6 == 0.8
 program is comppleted
